metrics/web/docs/features/searchable_dropdown/01_searchable_dropdown_design.md

```diff
@@ -0,0 +1,208 @@
+# Searchable dropdown design
+
+## TL;DR
+
+A `Searchable Dropdown` is a UI component that allows the user to search and select the items from the specific list of items. We can utilize the `Searchable Dropdown` instead of a regular `Dropdown` if the list is quite long, making it easier for the end-user to interact with the app, as he or she can start typing their request and immediately see all matching options.
+
+## Metrics Web Application
+
+To create a reusable and customizable widget, we should implement the base `SearchableDropdown` widget, that is then can be used for module-specific widget implementation. For that, we should use the [`selection_menu` package](https://pub.dev/packages/selection_menu), as it provides highly customizable components. Consider the following sub-section describing the `Selection Menu` package and its components needed to build the `SearchableDropdown` widget. 
```

Comment:
If we are considering using a 3rd party component - we might want to assess that it's of good quality and has acceptable test coverage. What do you think if we add a small section on that? I'm assuming while it's made well it lacks tests currently so we might want to add tests there and have this intention captured here.